### DISCUSSION TITLE
Define extra fields in a many to many relation with pivot table

### DISCUSSION THREAD

**Discussion by AlexandrosMelissas** [mikro-orm/mikro-orm#2251] [Q&A] [answered]
Is it possible to define extra fields in the pivot table that gets created in a many to many relation? Or do we have to define a new entity every time 3 or more fields exist in our m2m relation?

**Comment by B4nan** [comment, marked as answer]
You need to have a separate entity, otherwise it would be impossible to write or read such values. 

https://mikro-orm.io/docs/composite-keys/#use-case-3-join-table-with-metadata

**Comment by AlexandrosMelissas** [comment]
I understand now. Thank you.
